homeassistant/components/soma/__init__.py

```diff
@@ -0,0 +1,111 @@
+"""Support for Soma Smartshades."""
+import logging
+
+import voluptuous as vol
+from api.soma_api import SomaApi
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.typing import HomeAssistantType
+
+from homeassistant.const import CONF_HOST, CONF_PORT
+
+from .const import DOMAIN, HOST, PORT, API
+
+
+DEVICES = "devices"
+
+_LOGGER = logging.getLogger(__name__)
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {vol.Required(CONF_HOST): cv.string, vol.Required(CONF_PORT): cv.string}
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+SOMA_COMPONENTS = ["cover"]
+
+
+async def async_setup(hass, config):
+    """Set up the Soma component."""
+    if DOMAIN not in config:
+        return True
+
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            data=config[DOMAIN],
+            context={"source": config_entries.SOURCE_IMPORT},
+        )
+    )
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistantType, entry: ConfigEntry):
+    """Set up Soma from a config entry."""
+    hass.data[DOMAIN] = {}
+    hass.data[DOMAIN][API] = SomaApi(entry.data[HOST], entry.data[PORT])
+    devices = await hass.async_add_executor_job(hass.data[DOMAIN][API].list_devices)
+    hass.data[DOMAIN][DEVICES] = devices["shades"]
+
+    for component in SOMA_COMPONENTS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(entry, component)
+        )
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistantType, entry: ConfigEntry):
+    """Unload a config entry."""
+    return True
```

Comment:
For a future PR we can unload the cover platform here by forwarding the entry unloading to the cover component.